cmake/third_party/openfst.cmake

```diff
@@ -1,14 +1,25 @@
-cmake_minimum_required(VERSION 2.8.2)
-project(openfst-download NONE)
-
-include(ExternalProject)
-ExternalProject_Add(openfst
-    GIT_REPOSITORY https://github.com/kkm000/openfst
-    GIT_TAG 0bca6e76d24647427356dc242b0adbf3b5f1a8d9 # tag win/1.7.2.1
-    SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/openfst"
-    BINARY_DIR ""
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-    INSTALL_COMMAND ""
-    TEST_COMMAND ""
+cmake_minimum_required(VERSION 3.14)
+include(FetchContent)
+
+FetchContent_Declare(
+        openfst
+        GIT_REPOSITORY  https://github.com/kkm000/openfst
+        GIT_TAG         338225416178ac36b8002d70387f5556e44c8d05 # tag win/1.7.2.1

```

Comment:
tag win/1.7.2.1... does this mean it's specifically for Windows?

Reply:
the comment , I test on ubuntu and WSL(arch)

> Looks like a definite improvement, thanks! Does it build binaries as well? And did you test it on Linux, too?
> 
> Some recipes require FAR extensions of OpenFST (the configure switches in our tools/Makefile aren't exactly correct, I suspect). But this is not a show-stopper, we can take care of that next.

I have built the project on ubuntu and WSL (arch) using ninja build. The configure switches depend on the CMakeLists.txt in the OpenFST project.

Reply:
The tag is the newest commit of the winport branch not specifically for Windows, I forgot to remove the old comment, sorry.